src/Identity/Extensions.Stores/src/IdentityUserPasskey.cs

```diff
@@ -0,0 +1,85 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.AspNetCore.Identity;
+
+/// <summary>
+/// Represents a passkey credential for a user in the identity system.
+/// </summary>
+/// <remarks>
+/// See <see href="https://www.w3.org/TR/webauthn-3/#credential-record"/>.
+/// </remarks>
+/// <typeparam name="TKey">The type used for the primary key for this passkey credential.</typeparam>
+public class IdentityUserPasskey<TKey> where TKey : IEquatable<TKey>
+{
+    /// <summary>
+    /// Gets or sets the primary key of the user that owns this passkey.
+    /// </summary>
+    public virtual TKey UserId { get; set; } = default!;
+
+    /// <summary>
+    /// Gets or sets the credential ID for this passkey.
+    /// </summary>
+    public virtual byte[] CredentialId { get; set; } = [];
+
+    /// <summary>
+    /// Gets or sets the public key associated with this passkey.
+    /// </summary>
+    public virtual byte[] PublicKey { get; set; } = [];
+
+    /// <summary>
+    /// Gets or sets the friendly name for this passkey.
+    /// </summary>
+    public virtual string? Name { get; set; }
+
+    /// <summary>
+    /// Gets or sets the time this passkey was created.
+    /// </summary>
+    public virtual DateTimeOffset CreatedAt { get; set; }
+
+    /// <summary>
+    /// Gets or sets the signature counter for this passkey.
+    /// </summary>
+    public virtual uint SignCount { get; set; }
+
+    /// <summary>
+    /// Gets or sets the transports supported by this passkey.
+    /// </summary>
+    /// <remarks>
+    /// See <see href="https://www.w3.org/TR/webauthn-3/#enumdef-authenticatortransport"/>.
+    /// </remarks>
+    public virtual string[]? Transports { get; set; }
+
+    /// <summary>
+    /// Gets or sets whether the passkey has a verified user.
+    /// </summary>
+    public virtual bool IsUserVerified { get; set; }
+
+    /// <summary>
+    /// Gets or sets whether the passkey is eligible for backup.
+    /// </summary>
+    public virtual bool IsBackupEligible { get; set; }
+
+    /// <summary>
+    /// Gets or sets whether the passkey is currently backed up.
+    /// </summary>
+    public virtual bool IsBackedUp { get; set; }
```

Comment:
These flags are all present on the [authenticator data](https://www.w3.org/TR/webauthn-3/#sctn-authenticator-data) flags. We could just store the flags byte directly so that we don't introduce another schema change in the future, should we find ourselves wanting to store more flags.

Reply:
The specification states in the [Credential Backup State](https://www.w3.org/TR/webauthn-3/#sctn-credential-backup) section:
> "The value of the BE flag is set during authenticatorMakeCredential operation and MUST NOT change."

This means:
* The BE (Backup Eligible) flag must remain unchanged after Passkey creation.
* If BE was initially 0 (false), the BS (Backup State) flag must never be 1 (true).

Storing these flags in binary form would be impractical. The current implementation (keeping them as separate values) provides the right balance of correctness and usability.